modules/getComponents.js

```diff
@@ -8,7 +9,13 @@ function getComponentsForRoute(nextState, route, callback) {
 
   const getComponent = route.getComponent || route.getComponents
   if (getComponent) {
-    getComponent.call(route, nextState, callback)
+    const componentReturn = getComponent.call(route, nextState, callback)
+    if (isPromise(componentReturn))
+      componentReturn
+        // Try module.default first in case of System.import and Babel 6
+        .then(component => component.default || component)
```

Comment:
I feel like that should live in user space though. We don't have anything like this in the callback-based path. I do agree that this is friendlier, but I think philosophically it's not right here.

Also, webpack 2 can handle these things natively now... users on webpack 2 actually are using that native ES module support now that we have the `module` entry point.